src/main/views/viewManager.ts

```diff
@@ -112,44 +122,73 @@ export class ViewManager {
         this.configServers.forEach((server) => this.loadServer(server));
     }
 
+    /** Called when a new configuration is received
+     * Servers or tabs have been added or edited. We need to
+     * close, open, or reload tabs, taking care to reuse tabs and
+     * preserve focus on the currently selected tab. */
     reloadConfiguration = (configServers: TeamWithTabs[]) => {
-        this.configServers = configServers.concat();
-        const oldviews = this.views;
+        const focusedTuple: TabTuple | undefined = this.views.get(this.currentView as string)?.tuple;
+
+        const current: Map<TabTuple, MattermostView> = new Map();
+        for (const x of this.views.values()) {
+            current.set(x.tuple, x);
+        }
+
+        const views: Map<TabTuple, MattermostView> = new Map();
+        const closed: Map<TabTuple, {srv: MattermostServer; tab: Tab; name: string}> = new Map();
+
+        const sortedTabs = configServers.flatMap((x) => x.tabs.sort(by((x) => x.order)).map((t) => duad(x, t)));
```

Comment:
We can simplify and standardize this without needing these functional helper functions, see my suggestion:
```suggestion
        const sortedTabs = configServers.flatMap((x) => x.tabs.sort((a, b) => a.order - b.order).map((t) => [x, t]);
```